pkg/machinefilters/machine_filters.go

```diff
@@ -78,7 +79,7 @@ func MatchesKubernetesVersion(kubernetesVersion string) Func {
 		if machine.Spec.Version == nil {
 			return false
 		}
-		return *machine.Spec.Version == kubernetesVersion
+		return strings.TrimPrefix(*machine.Spec.Version, "v") == strings.TrimPrefix(kubernetesVersion, "v")
```

Comment:
this is OK to start with, but will need more iterating when we are doing the in-place upgrades work.